src/pages/workspace/WorkspaceInviteMessagePage.js

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Pressable, View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import Str from 'expensify-common/lib/str';
+import lodashGet from 'lodash/get';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
+import Navigation from '../../libs/Navigation/Navigation';
+import styles from '../../styles/styles';
+import compose from '../../libs/compose';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as Policy from '../../libs/actions/Policy';
+import TextInput from '../../components/TextInput';
+import MultipleAvatars from '../../components/MultipleAvatars';
+import CONST from '../../CONST';
+import * as Link from '../../libs/actions/Link';
+import Text from '../../components/Text';
+import withPolicy, {policyPropTypes, policyDefaultProps} from './withPolicy';
+import * as ReportUtils from '../../libs/ReportUtils';
+import ROUTES from '../../ROUTES';
+import * as Localize from '../../libs/Localize';
+import Form from '../../components/Form';
+
+const personalDetailsPropTypes = PropTypes.shape({
+    /** The login of the person (either email or phone number) */
+    login: PropTypes.string.isRequired,
+
+    /** The URL of the person's avatar (there should already be a default avatar if
+    the person doesn't have their own avatar uploaded yet) */
+    avatar: PropTypes.string.isRequired,
+
+    /** This is either the user's full name, or their login if full name is an empty string */
+    displayName: PropTypes.string.isRequired,
+});
+
+const propTypes = {
+
+    /** All of the personal details for everyone */
+    personalDetails: PropTypes.objectOf(personalDetailsPropTypes).isRequired,
+
+    invitedMembersDraft: PropTypes.arrayOf(PropTypes.string).isRequired,
+
+    /** URL Route params */
+    route: PropTypes.shape({
+        /** Params from the URL path */
+        params: PropTypes.shape({
+            /** policyID passed via route: /workspace/:policyID/invite-message */
+            policyID: PropTypes.string,
+        }),
+    }).isRequired,
+
+    ...policyPropTypes,
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = policyDefaultProps;
+
+class WorkspaceInviteMessagePage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.onSubmit = this.onSubmit.bind(this);
+        this.validate = this.validate.bind(this);
+        this.openPrivacyURL = this.openPrivacyURL.bind(this);
+        this.state = {
+            welcomeNote: this.getWelcomeNote(),
+        };
```

Comment:
Instead of keeping this state, can we just pass the welcome message as the `defaultValue` for the `TextInput`, e.g.

```
<TextInput
  ...props
  defaultValue={this.props.translate('workspace.inviteMessage.welcomeNote', {
            workspaceName: this.props.policy.name,
        })};
/>
```

Reply:
Form.js defaultValue takes the value only during first time - https://github.com/Expensify/App/blob/main/src/components/Form.js#L255
So if the language changes when we are already in this message, the default message won't be translated.


Do we want to change Form.js component in a way to support this usecase?

Reply:
That's a good point and IMO a bug with the TextInput itself. I think that we'd want to update the TextInput default value when the locale changes in case we have provided a defaultValue that needs to be translated. We do something similar for [Picker](https://github.com/Expensify/App/blob/5cba7218adbc0d4f8383b312beb7b9d90d2f8898/src/components/Picker.js#L127).

Reply:
@luacmartins  I am not sure if understood the above response correctly. I see we did a similar fix Picker component, do we want to do a similar to fix to TextInput/Form itself as part of this PR, and update the message field in this particular page accordingly? Or can we do it as a separate improvement fix as its already working now in old way or else it Seems we would be bloating this PR. 
**I can even volunteer to take it up as a separate fix on TextInput/Form.js as well and modify the invite message page accordingly**
But its just my opinion, but I am open to fix it either ways.

cc: @mollfpr

Reply:
@abdulrahuman5196 Here's some basic idea for a similar implementation of the Picker component.

```js
import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import TextInput from './TextInput';
import * as Localize from '../libs/Localize';
import withLocalize, {withLocalizePropTypes} from './withLocalize';
import compose from '../libs/compose';
import withPolicy from '../pages/workspace/withPolicy';

const propTypes = {
    inputID: PropTypes.string,
    onInputChange: PropTypes.func.isRequired,

    ...withLocalizePropTypes,
};

const defaultProps = {
    inputID: undefined,
};

class InviteMessageTextInput extends PureComponent {
    constructor(props) {
        super(props);

        this.onInputChange = this.onInputChange.bind(this);
    }

    componentDidUpdate(prevProps) {
        if (!(prevProps.preferredLocale !== this.props.preferredLocale
            && this.props.value === Localize.translate(prevProps.preferredLocale, 'workspace.inviteMessage.welcomeNote', {workspaceName: this.props.policy.name}))
        ) {
            return;
        }
        this.onInputChange(this.props.translate('workspace.inviteMessage.welcomeNote', {
            workspaceName: this.props.policy.name,
        }));
    }

    onInputChange(value) {
        if (!this.props.inputID) {
            return;
        }
        this.props.onInputChange(value);
    }

    render() {
        return (
            <TextInput
                // eslint-disable-next-line react/jsx-props-no-spreading
                {...this.props}
                onInputChange={this.onInputChange}
            />
        );
    }
}

InviteMessageTextInput.propTypes = propTypes;
InviteMessageTextInput.defaultProps = defaultProps;

export default compose(
    withLocalize,
    withPolicy,
)(InviteMessageTextInput);

```

We override the `onInputChange` from the `Form.js` component and pass the translated welcome message when the locale changes.

Reply:
@mollfpr  Thank you for the reference. I agree on the thought process here. But my concern is, this seems to be an generic issue, where we need to support **"Changing default value of Text Input with locale changes/ or any other reason to update default value which is shown"**.

 I was thinking of providing this support in TextInput component itself, like exposing a param like 'shouldUseNewDefaultValueOnUpdates' similar to existing shouldSaveDraft and update the input internally. So that we would only need code like 

```
<TextInput
  ...props
shouldUseNewDefaultValueOnUpdates

// expected to provide new values when locale changes
  defaultValue={this.props.translate('workspace.inviteMessage.welcomeNote', {
            workspaceName: this.props.policy.name,
        })};
/>
```

Reply:
Just posting my opinions as its better to gather more information and align on the path forward before implementation as we found this defaultValue issue recently.

Reply:
I agree that we should fix this in the TextInput itself and we could probably do that as a follow up to separate concerns. I don't think we need another prop though. Could we just add the check below to ComponentDidUpdate in TextInput?

```
componentDidUpdate(prevProps) {
    if (prevProps.defaultValue !== this.props.defaultValue) {
        this.setState({value: this.props.value || this.props.defaultValue || '' });
    }
}
```

Reply:
Agree on the above. 
The prop was just proposed to add as an additional control on the behaviour, as a component can use the current behaviour(where defaultValue doesn't update) without any change if they don't send the prop. And only the expected components will use this new behaviour.
Just wanted to be safe as we don't break any existing experiences. 

But its still fine if we don't create a prop itself, just that we have to make sure none of the current experiences using TextInput breaks.

Reply:
```
componentDidUpdate(prevProps) {
    if (prevProps.defaultValue !== this.props.defaultValue) {
        this.setState({value: this.props.value || this.props.defaultValue || '' });
    }
}
```

I'm not sure if the above snippet will work on updating the value with updated `defaultValue`. The `this.props.value` will apply first from the `Form.js`. Maybe I miss something 🤷

Reply:
but we'd want any values to take precedence over the `defaultValue`, no? Or are you saying that the first `defaultValue` will be stored as value and then just override subsequent updates to `defaultValue`?

Reply:
Yes, any values should take precedence over the `defaultValue`. I mean, `this.props.defaultValue` will not be used `this.setState({value: this.props.value || this.props.defaultValue || '' });` because we have `this.props.value` from the `Form.js`.

Reply:
Ah I see what you are saying now. Hmm yea, I think we should fix this in Form instead of TextInput, but I think we should do that as a separate issue/PR.

Reply:
We have this PR https://github.com/Expensify/App/pull/15772 that adds an option to force the use of `defaultValue`, but I'm not sure if it can solve our current concern.

Reply:
> Yes, any values should take precedence over the defaultValue. I mean, this.props.defaultValue will not be used this.setState({value: this.props.value || this.props.defaultValue || '' }); because we have this.props.value from the Form.js.

Got it, Its because we don't send 'defaultValue' instead just 'value' to TextInput. In that case we can solve it in Form itself. This case should be generic for component to update when default value changes

>We have this PR https://github.com/Expensify/App/pull/15772 that adds an option to force the use of defaultValue, but I'm not sure if it can solve our current concern.

It forcefully uses the defaultValue. It won't solve our case as we want to use the defaultValue only when there is no user provided input.

Reply:
@luacmartins  @mollfpr  I think we should revive this thread to think about adding support for Text inside Form to support changing defaultValue. 

Another issue which might have used this support if it where available(not entirely same but just quoting)
https://github.com/Expensify/App/issues/19460

Reply:
@abdulrahuman5196 maybe you can suggest that in the issue you linked?

Reply:
Ideally feel free to bring this up in Slack

Reply:
Started a thread to see if need this support or not
https://expensify.slack.com/archives/C01GTK53T8Q/p1685436048476259